iceoryx_posh/include/iceoryx_posh/internal/popo/base_subscriber.inl

```diff
@@ -158,7 +161,15 @@ BaseSubscriber<T, Subscriber, port_t>::enableEvent(iox::popo::TriggerHandle&& tr
                                                    [[gnu::unused]] const SubscriberEvent subscriberEvent) noexcept
 {
     m_trigger = std::move(triggerHandle);
-    m_port.setConditionVariable(m_trigger.getConditionVariableData());
+    if (m_trigger.doesContainEventVariable())
```

Comment:
Exactly this will be addressed in the next Pull Request. If you (@elBoberido & @mossmaurice ) insist on it please let me know here but be aware that this Pull Request will then be much larger.